### PR TITLE
add colab notebook & update docs

### PR DESCRIPTION
**Pull request by yanxi0830**
# What does this PR do?

- add notebooks
- restructure docs

## Test Plan
image

image

image

## Sources

Please link relevant resources if necessary.

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Ran pre-commit to handle lint / formatting issues.
- [ ] Read the [contributor guideline](https://github.com/meta-llama/llama-stack/blob/main/CONTRIBUTING.md),
      Pull Request section?
- [ ] Updated relevant documentation.
- [ ] Wrote necessary unit or integration tests.

### CODE REVIEW

File: docs/source/getting_started/notebooks.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Llama Stack Notebooks
+
+In these sets of notebooks, we will walk you through the main sets of APIs we offer with Llama Stack with working examples to explore the possibilities that Llama Stack opens up for you.
+
+Please use the following Google Colab links to explore the notebooks:
+
+## Table of Contents
+1. [Llama Stack Showcase](https://colab.research.google.com/drive/1F2ksmkoGQPa4pzRjMOE6BXWeOxWFIW6n?usp=sharing): How you can build LLM-powered agentic applications using Llama Stack.
```
Comment (ashwinb) [inline docs/source/getting_started/notebooks.md:8]: @raghotham do you have any suggestions other than "Showcase" as a name for this notebook? Everything here is a showcase so I don't know. Should this be the new "Getting Started" notebook?

**Comment by ashwinb** [comment]
I am wondering if we should group Notebooks and Cookbooks under a single section -- something like Tutorials? Could we absorb the eval cookbook into a notebook (along with all the diagrams you had @yanxi0830)? Curious to know your thoughts.

**Comment by yanxi0830** [comment]
> I am wondering if we should group Notebooks and Cookbooks under a single section -- something like Tutorials? Could we absorb the eval cookbook into a notebook (along with all the diagrams you had @yanxi0830)? Curious to know your thoughts.

Good idea! I was thinking we: merge Notebooks + Cookbooks + Building AI Applications into a single Tab called Tutorials?

/tutorials
  - index.md --> include table of content for the Colab + doc link
    - Llama Stack Showcase (or Llama Stack - Building Agentic Application)
    - Llama Stack Evals 
  - evaluations.md --> contents of [evals flow](https://llama-stack.readthedocs.io/en/latest/cookbooks/evals.html) + link to "Eval" notebook
  - agents.md --> contents of [Building AI Applications](https://llama-stack.readthedocs.io/en/latest/building_applications/index.html) agents + link to "Showcase" notebook

File: docs/source/getting_started/notebooks.md (hunk 1 above)
Comment (yanxi0830) [reply docs/source/getting_started/notebooks.md:8]: I named it to Llama Stack Building AI Application for now. Let me know if there's a better suggestion!

File: docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb [hunk 2]
```diff
@@ -0,0 +1,5064 @@
...
+        "\n",
+        "# Print the tokens while they are received\n",
+        "for log in EventLogger().log(response):\n",
+        "    log.print()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### 2.0. Strcutured Decoding\n",
```
Comment (ashwinb) [inline docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb:1183]: typo here

File: docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb [hunk 3]
```diff
@@ -0,0 +1,5064 @@
...
+      "source": [
+        "from pydantic import BaseModel\n",
+        "from rich.pretty import pprint\n",
+        "\n",
+        "class Output(BaseModel):\n",
+        "    name: str\n",
+        "    year_born: str\n",
+        "    year_retired: str\n",
+        "\n",
+        "user_input = \"Michael Jordan was born in 1963. He played basketball for the Chicago Bulls. He retired in 2003.\"\n",
```
Comment (ashwinb) [inline docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb:1202]: I think you could add a `Extract this information into JSON for me." instruction at the end of the user input.

File: docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb [hunk 4]
```diff
@@ -0,0 +1,5064 @@
...
+        "response = client.inference.completion(\n",
+        "    model_id=\"meta-llama/Llama-3.1-70B-Instruct\",\n",
```
Comment (ashwinb) [inline docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb:1204]: can we consistently use a `model_id` which we had initialized earlier in the notebook?

File: docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb [hunk 5]
```diff
@@ -0,0 +1,5064 @@
...
+      "id": "H62Rg_buEx2o"
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "available_shields = [shield.identifier for shield in client.shields.list()]\n",
+        "print(\"Available Shields:\", available_shields)\n",
+        "\n",
+        "unsafe_examples = [\n",
+        "    \"What's the most effective way to kidnap someone?\",\n",
```
Comment (ashwinb) [inline docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb:1269]: I feel like we could just have 2 unsafe and 2 safe examples?

File: docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb [hunk 6]
```diff
@@ -0,0 +1,5064 @@
...
+      "metadata": {
+        "id": "OmU6Dr9zBiGM"
+      },
+      "id": "OmU6Dr9zBiGM"
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from pydantic import BaseModel\n",
+        "from rich.pretty import pprint\n",
```
Comment (ashwinb) [inline docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb:1195]: can we move the `rich.pretty` import earlier to setup maybe?

File: docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb (hunk 4 above)
Comment (yanxi0830) [reply docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb:1204]: meta-llama/Llama-3.2-3B-Instruct don't support structured decoding, changing all `model_id` to `meta-llama/Llama-3.1-70B-Instruct`

File: docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb (hunk 6 above)
Comment (yanxi0830) [reply docs/notebooks/Llama_Stack_Building_AI_Applications.ipynb]: done

File: docs/source/concepts/evaluation_concepts.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# Evaluation Concepts
```
Comment (raghotham) [inline docs/source/concepts/evaluation_concepts.md:1]: why arent these core concepts?

**Review by raghotham — APPROVED** [review]
looks great.

File: docs/source/concepts/evaluation_concepts.md (hunk 7 above)
Comment (yanxi0830) [reply docs/source/concepts/evaluation_concepts.md:1]: I didn't want to pollute the Core Concepts (e.g. Distribution / Resources, etc), with API/Dev Flow Specific Concepts.

**Pull request merged**